elixir/apps/domain/lib/domain/config/definitions.ex

```diff
@@ -459,7 +459,6 @@ defmodule Domain.Config.Definitions do
       :string, changeset, key ->
         changeset
         |> Domain.Validator.trim_change(key)
```

Comment:
Ah, sorry I wasn't thinking through this clearly when we discussed on Slack. So I think it might actually make sense to keep this validation but expand it to include URIs as well, e.g.:

```suggestion
        |> Domain.Validator.trim_change(key)
        |> Domain.Validator.validate_fqdn_or_uri(key)
```

That should cover DoH and DoT use cases for now, and we can expand it later when more use cases emerge.

Reply:
Just adding to Jamils comment we should add a helper for that:
```
Validator.validate_one_of(changeset, [&validate_fqdn/2, &validate_uri/2])
```
which will apply validation functions until one of them is valid, or accumulates all errors and returns them for a field otherwise.

Reply:
I implemented the `validate_one_of` method in one of the later commits.